### PR TITLE
[HEVCe] Support Dynamic Slice Size in refactor code

### PR DESCRIPTION
**Pull request by JeremyShang**
To enable dynamic slice size, use MaxSliceSize to specify maximum slice size in bytes.

Signed-off-by: Jeremy Shang <jeremy.shang@intel.com>

### CODE REVIEW

File: _studio/mfx_lib/encode_hw/hevc/linux/base/hevcehw_base_va_packer_lin.cpp [hunk 1]
```diff
@@ -635,6 +637,20 @@ void VAPacker::InitAlloc(const FeatureBlocks& /*blocks*/, TPushIA Push)
...
         });
+        cc.AddPerPicMiscData[VAEncMiscParameterTypeMaxSliceSize].Push([](
+            CallChains::TAddMiscData::TExt
+            , const StorageR& global
+            , const StorageR&
+            , std::list<std::vector<mfxU8>>& data)
+        {
+            const auto& par = Base::Glob::VideoParam::Get(global);
+            const mfxExtCodingOption2& CO2 = ExtBuffer::Get(par);
+            auto& maxSliceSize_param = AddVaMisc<VAEncMiscParameterMaxSliceSize>(VAEncMiscParameterTypeMaxSliceSize, data);
```
Comment (vilichev) [inline _studio/mfx_lib/encode_hw/hevc/linux/base/hevcehw_base_va_packer_lin.cpp:648]: I think we don't need to submit VAEncMiscParameterMaxSliceSize if CO2.MaxSliceSize == 0
What about reset, do we need to resubmit it on reset?

Comment (JeremyShang) [reply _studio/mfx_lib/encode_hw/hevc/linux/base/hevcehw_base_va_packer_lin.cpp]: I don't think we need to resubmit it on reset.

**Review by Tianhaol — APPROVED** [review]

File: _studio/mfx_lib/encode_hw/hevc/linux/base/hevcehw_base_va_packer_lin.cpp (hunk 1 above)
Comment (vilichev) [reply _studio/mfx_lib/encode_hw/hevc/linux/base/hevcehw_base_va_packer_lin.cpp]: Do we have such restriction in driver so we cannot reset CO2.MaxSliceSize to a new value? Seems such functionality is present in legacy code.

Comment (JeremyShang) [reply _studio/mfx_lib/encode_hw/hevc/linux/base/hevcehw_base_va_packer_lin.cpp]: Thanks @vilichev , we don't have such restriction in driver. So I upload a new commit to add it in reset.

**Review by vilichev — APPROVED** [review]

**Review by Tianhaol — APPROVED** [review]

**Review by saosipov — APPROVED** [review]

**Review by dmitryermilov — APPROVED** [review]

**Pull request merged**